### PR TITLE
Journalist can login

### PR DESCRIPTION
**Pull request by mantanquzzmo**
[PT-Link](https://www.pivotaltracker.com/story/show/170673056)

- Adds login functionality to client with j-tockauth

### CODE REVIEW

File: cypress/integration/journalistCanLogin.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    });
+    cy.visit("/");
+    cy.get("#loginButton").click();
+    cy.get("#login").within(() => {
+      cy.get("#email").type("user@mail.com");
+      cy.get("#password").type("wrongpassword");
+      cy.get("button")
+        .contains("Submit")
+        .click();
+    });
```
Comment (sundqvistjohan) [inline cypress/integration/journalistCanLogin.js:36]: There is no assertion here, should see the response error message?

Comment (sundqvistjohan) [reply cypress/integration/journalistCanLogin.js:36]: Okay I see what you're doing with the alert but a quick google suggests there is ways to test an alert box with cypress (e.g. https://stackoverflow.com/questions/51795306/how-can-we-test-the-alert-and-text-inside-is-displaying-using-cypress-io-js-auto)

File: src/components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  switch (true) {
+    case displayLoginButton && !props.authenticated:
+      loginButton = (
+        <button id="loginButton" onClick={() => setDisplayLoginButton(false)}>
+          Login
+        </button>
+      );
+      break;
+    case !displayLoginButton && !props.authenticated:
+      loginButton = (
```
Comment (sundqvistjohan) [inline src/components/Login.jsx]: It's a bit weird to call this `loginButton` as it's only a login button in the first out of the three cases

File: cypress/integration/journalistCanLogin.js (hunk 1 above)
Comment (tochman) [reply cypress/integration/journalistCanLogin.js:36]: I would need to see an assertion though

File: src/components/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    event.preventDefault();
+    auth
+      .signIn(event.target.email.value, event.target.password.value)
+      .then(userDatas => {
+        props.changeAuth(true);
+        setLoginMessage(`Logged in as: ${userDatas.data.email}`);
+      })
+      .catch(error => {
+        alert("Invalid login credentials");
+        window.location.reload();
```
Comment (tochman) [inline src/components/Login.jsx]: Not sure if we should have to reload the page like this.  Not at all

**Review by tochman — CHANGES_REQUESTED** [review]

File: cypress/integration/journalistCanLogin.js (hunk 1 above)
Comment (mantanquzzmo) [reply cypress/integration/journalistCanLogin.js:36]: done, changed so that the message handles the error

File: src/components/Login.jsx (hunk 3 above)
Comment (mantanquzzmo) [reply src/components/Login.jsx]: Removed it.

File: src/components/Login.jsx (hunk 2 above)
Comment (mantanquzzmo) [reply src/components/Login.jsx]: changed name to loginFunction

File: cypress/integration/journalistCanLogin.js (hunk 1 above)
Comment (mantanquzzmo) [reply cypress/integration/journalistCanLogin.js:36]: also added assertion

File: src/components/Login.jsx (hunk 3 above)
Comment (sundqvistjohan) [reply src/components/Login.jsx]: Good, from a user perspective I dislike having to re-enter my email/username after using the wrong password

**Review by tochman — APPROVED** [review]
Looks good

**Pull request merged**